portal-impl/src/com/liferay/portal/security/permission/AdvancedPermissionChecker.java

```diff
@@ -444,7 +444,8 @@ protected boolean doCheckPermission(
 		}
 
 		long[] roleIds = PermissionCacheUtil.getUserGroupRoleIds(
-			userId, groupId);
+			userId,
+			GetterUtil.getLong(StagingPermissionChecker.getGroupId(), groupId));
```

Comment:
Save value as teamGroupId or something.

Reply:
what about currentGroupId? In this way it can be reused for other cases in the future. TeamGroupId is also good since team should be a unique exception.